src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+    Animal.checkIsAlive();
+  }
 }
 
+Animal.alive = [];
+
+Animal.checkIsAlive = function() {
```

Comment:
```suggestion
Animal.removeDeadAnimals = () => {
```
The method must clearly describe its functionality.